src/test/kotlin/com/atlassian/performance/tools/lib/Apdex.kt

```diff
@@ -12,17 +12,30 @@ class Apdex {
     fun score(
         metrics: List<ActionMetric>
     ): Double {
-        return metrics
-            .filter { it.result == ActionResult.OK }
-            .map { score(it) }
+        return average(weight(metrics
+            .filter { it.result == ActionResult.OK }))
+    }
+
+    fun average(
```

Comment:
`score` filters by `OK`, but `average` and `weight` don't. This can easily lead to non-obvious interpretations further down the pipeline. I'd keep the consistency one way or another.

Taking a step back - what *should* be the Apdex relation to errors? Apdex is a measure of user experience. In my book, getting an error is definitely not a _satisfactory_ experience and probably not even _tolerable_. It's _frustrating_.

We can filter errors out and rely on other metrics to punish errors (e.g. error rate).
We can also leave errors in and make Apdex punish errors by itself.

Reply:
PS. This is also why it's worth importing parts of https://github.com/atlassian/jira-hardware-exploration/commit/1035cf16620d4e3a7d220e01f103e638864ac724, especially `enum class Experience`, because it reminds that Apdex is all about UX.

Reply:
I've pulled in the Experience, but I think the definition of Apdex is a wider question :)

Reply:
I have pulled on the Experience, but while I agree with you I think changing the Apdex definition is beyond this scope.

I have not changed the ActionResult filter because the text explicitly needs unfiltered metrics, but I have renamed the functions to hopefully be more explicit.